demo_nodes_cpp/CMakeLists.txt

```diff
@@ -77,51 +77,149 @@ add_dependencies(services_library)
 add_dependencies(parameters_library)
 add_dependencies(topics_library)
 
-rclcpp_components_register_node(timers_library
-  PLUGIN "demo_nodes_cpp::OneOffTimerNode"
-  EXECUTABLE one_off_timer)
-rclcpp_components_register_node(timers_library
-  PLUGIN "demo_nodes_cpp::ReuseTimerNode"
-  EXECUTABLE reuse_timer)
-
-rclcpp_components_register_node(services_library
-  PLUGIN "demo_nodes_cpp::ServerNode"
-  EXECUTABLE add_two_ints_server)
-rclcpp_components_register_node(services_library
-  PLUGIN "demo_nodes_cpp::ClientNode"
-  EXECUTABLE add_two_ints_client_async)
-
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ListParameters"
-  EXECUTABLE list_parameters)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ParameterBlackboard"
-  EXECUTABLE parameter_blackboard)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::SetAndGetParameters"
-  EXECUTABLE set_and_get_parameters)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::ParameterEventsAsyncNode"
-  EXECUTABLE parameter_events_async)
-rclcpp_components_register_node(parameters_library
-  PLUGIN "demo_nodes_cpp::EvenParameterNode"
-  EXECUTABLE even_parameters_node)
-
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::Talker"
-  EXECUTABLE talker)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::SerializedMessageTalker"
-  EXECUTABLE talker_serialized_message)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::Listener"
-  EXECUTABLE listener)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::SerializedMessageListener"
-  EXECUTABLE listener_serialized_message)
-rclcpp_components_register_node(topics_library
-  PLUGIN "demo_nodes_cpp::ListenerBestEffort"
-  EXECUTABLE listener_best_effort)
+rclcpp_components_register_nodes(
+  PLUGINS
+    "demo_nodes_cpp::ReuseTimerNode"
+    "demo_nodes_cpp::OneOffTimerNode"
+  FROM timers_library
+)
+
+rclcpp_components_add_executable(one_off_timer
+  PLUGINS "demo_nodes_cpp::ReuseTimerNode"
```

Comment:
Yes, I fully agree.  Actually, we use terms like `composable nodes`, `components`, `plugins` and `node plugins` interchangeably in many places and that's nuts. Components sounds reasonable to me to standardize to.

Reply:
Agree with standardizing on `component`

Reply:
Hmm, yeah, I dunno, components is just such a generic term. I'd prefer `COMPOSABLE_NODES` or otherwise `NODE_COMPONENTS`, but I if you guys really feel like `COMPONENTS` is right, then I'd defer to that.

Reply:
Either way I think it would be best for us to get our terminology straight "composable node" versus "node component" and use it everywhere. We could also use "component", but again I think that has too much overlap potential with existing computer science tools/ideas.

Reply:
Composable node is also fine, node components sound like there are other kinds of components (will there ever be?)

Reply:
> node components sound like there are other kinds of components (will there ever be?)

No idea.

I like "composable node" because it can be more naturally extended in English, e.g. "composable lifecycle node" or "composable light-weight node" or whatever. I guess you can do the same with "node component", but it feels more natural to me the other way at least.

Reply:
+1 for standardizing the term(s). The way I've been thinking about it is that there are *composable nodes* and one way we can compose them by making *components*. Who knows, maybe in the future there will exist other mechanisms other than *(node) components* for composing nodes.

My two cents.